rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -1358,8 +1358,27 @@ impl ChunkInstructions {
 
             while rows_needed > 0 || need_preamble {
                 let chunk = &rep_index.blocks[block_index];
-                let rows_avail = chunk.starts_including_trailer - to_skip;
-                debug_assert!(rows_avail > 0);
+                let rows_avail = chunk.starts_including_trailer.saturating_sub(to_skip);
+
+                // Handle blocks that are entirely preamble (rows_avail = 0)
+                // These blocks have no rows to take but may have a preamble we need
+                if rows_avail == 0 {
```

Comment:
I think this needs to be `rows_avail == 0 && to_skip == 0`.  If we need to skip even 1 row then we are not looking for the preamble.  That being said...I'm not sure if it is possible that `rows_avail == 0 && to_skip != 0` because otherwise we would have skipped the block in the binary search.  Still, might not hurt to be explicit.